server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/LoadReportAnalysisMetadataHolderStep.java

```diff
@@ -88,6 +103,43 @@ private void checkProjectKeyConsistency(ScannerReport.Metadata reportMetadata) {
     }
   }
 
+  private void checkOrganizationKeyConsistency(ScannerReport.Metadata reportMetadata, Organization organization) {
+    String organizationKey = reportMetadata.getOrganizationKey();
+    String resolveReportOrganizationKey = resolveReportOrganizationKey(organizationKey);
+    if (!resolveReportOrganizationKey.equals(organization.getKey())) {
+      if (reportBelongsToDefaultOrganization(organizationKey)) {
+        throw MessageException.of(format(
+          "Report does not specify an OrganizationKey but it has been submitted to another organization (%s) than the default one (%s)",
```

Comment:
Message does target the end user but the name of the field in the report is organizationKey. It will make it easier to this user if the name of the field with a wrong value is accurately reported, don't you think?